File: idea-plugin/src/github/jk1/smtpidea/actions/ConfigureAction.java
package github.jk1.smtpidea.actions;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.DialogWrapper;
import com.intellij.openapi.util.IconLoader;
import github.jk1.smtpidea.ui.ConfigurationDialog;

/**
 * Launches plugin configuration dialog.
 * If configuration has been changed running SMTP server is restarted.
 *
 * @author Evgeny Naumenko
 */
public class ConfigureAction extends AnAction {

    public ConfigureAction() {
        super("Configure", "Description", IconLoader.getIcon("/general/settings.png"));
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public void actionPerformed(final AnActionEvent anActionEvent) {
        final Project project = anActionEvent.getProject();
        if (project != null) {
            DialogWrapper dialog = new ConfigurationDialog(anActionEvent.getProject());
            dialog.show();
        }
    }

}

File: src/main/kotlin/github/jk1/smtpidea/actions/ConfigureAction.kt
package github.jk1.smtpidea.actions

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import github.jk1.smtpidea.ui.ConfigurationDialog
import github.jk1.smtpidea.Icons

/**
 * Launches plugin configuration dialog.
 * If configuration has been changed, then running server is restarted.
 */
object ConfigureAction : AnAction("Configure", "Open mail server configuration dialog", Icons.SETTINGS){

    override fun actionPerformed(p0: AnActionEvent?) {
        val project = p0?.getProject()
        if (project != null){
            ConfigurationDialog(project).show()
        }
    }
}